concordium-consensus/src/Concordium/KonsensusV1/SkovMonad.hs

```diff
@@ -661,20 +708,26 @@ migrateSkovV1 ::
 migrateSkovV1 regenesis migration gsConfig@GlobalStateConfig{..} oldPbsc oldBlockState bakerCtx handlerCtx unliftSkov migrateTT migratePTT = do
     pbsc@PersistentBlockStateContext{..} <- newPersistentBlockStateContext gsConfig
     logEvent GlobalState LLDebug "Migrating existing global state."
-    newInitialBlockState <- flip runBlobStoreT oldPbsc . flip runBlobStoreT pbsc $ do
-        newState <- migratePersistentBlockState migration $ hpbsPointers oldBlockState
-        hashBlockState newState
+    let newInitialBlockState :: InitMonad pv (HashedPersistentBlockState pv)
+        newInitialBlockState = do
+            flip runBlobStoreT oldPbsc . flip runBlobStoreT pbsc $ do
+                newState <- migratePersistentBlockState migration $ hpbsPointers oldBlockState
+                hashBlockState newState
     let
         initGS :: InitMonad pv (SkovData pv)
         initGS = do
-            stateRef <- saveBlockState newInitialBlockState
-            chainParams <- getChainParameters newInitialBlockState
-            genEpochBakers <- genesisEpochBakers newInitialBlockState
-            let genMeta = regenesisMetadata (getHash newInitialBlockState) regenesis
+            newState <- newInitialBlockState
+            stateRef <- do
+                ref <- saveBlockState newState
+                saveAccounts newState
```

Comment:
Why is this necessary? If this is migration from the old consensus aren't all accounts already stored?

Reply:
Yes, this shouldn't be necessary as you correctly point out the account map functions across protocol (and consensus versions).